testgen/src/tgen_sieve.erl

```diff
@@ -0,0 +1,40 @@
+-module('tgen_sieve').
+
+-behaviour(tgen).
+
+-export([
+    available/0,
+    generate_test/2
+]).
+
+-spec available() -> true.
+available() ->
+    true.
+
+generate_test(N, #{description := Desc, expected := Exp, property := Prop, input := #{limit := Limit}}) ->
+    TestName = tgen:to_test_name(N, Desc),
+    Property = tgen:to_property_name(Prop),
+
+    Fn = tgs:simple_fun(TestName, [
+        tgs:assign(tgs:var("Expected"), tgs:raw(format_list(Exp))),
+        tgs:call_macro("assertEqual", [
+            tgs:var("Expected"),
+            tgs:call_fun("lists:sort", [
+                tgs:call_fun("sieve:" ++ Property, [
+                    tgs:value(Limit)])])])]),
+
+    {ok, Fn, [{Property, ["Limit"]}]}.
+
+format_list([]) -> "[]";
+format_list(List) when length(List)=<10 ->
+    "["++lists:flatten(lists:join(", ", [io_lib:format("~B", [N]) || N <- List]))++"]";
+format_list(List) ->
+    MaxLen=length(integer_to_list(lists:max(List))),
+    List1=[io_lib:format("~"++integer_to_list(MaxLen)++"B", [N]) || N <- lists:sort(List)],
```

Comment:
Can't you use `io_lib:fortmat("~*B", [MaxLen, N])`? I consider that a little bit more readable than to manipulating the format string via list concatenation and function calls.

Reply:
Good find, I didn't know that was possible. Will change accordingly.